crates/re_renderer/examples/picking.rs

```diff
@@ -0,0 +1,205 @@
+use itertools::Itertools as _;
+use rand::Rng;
+use re_renderer::{
+    view_builder::{Projection, TargetConfiguration, ViewBuilder},
+    Color32, PickingLayerId, PickingLayerInstanceId, PointCloudBuilder, RenderContext,
+    ScheduledPickingRect, Size,
+};
+
+mod framework;
+
+struct Picking {
+    points_positions: Vec<glam::Vec3>,
+    points_radii: Vec<Size>,
+    points_colors: Vec<Color32>,
+    points_picking_ids: Vec<PickingLayerInstanceId>,
+
+    scheduled_picking_rects: Vec<ScheduledPickingRect>,
+
+    picking_position: glam::UVec2,
+}
+
+fn random_color(rnd: &mut impl rand::Rng) -> Color32 {
+    ecolor::Hsva {
+        h: rnd.gen::<f32>(),
+        s: rnd.gen::<f32>() * 0.5 + 0.5,
+        v: rnd.gen::<f32>() * 0.5 + 0.5,
+        a: 1.0,
+    }
+    .into()
+}
+
+impl Picking {
+    #[allow(clippy::unused_self)]
+    fn handle_incoming_picking_data(&mut self, re_ctx: &mut RenderContext, _time: f32) {
+        re_ctx
+            .gpu_readback_belt
+            .lock()
+            .receive_data(|data, identifier| {
+                if let Some(index) = self
+                    .scheduled_picking_rects
+                    .iter()
+                    .position(|s| s.identifier == identifier)
+                {
+                    let picking_rect_info = self.scheduled_picking_rects.swap_remove(index);
+
+                    // TODO(andreas): Move this into a utility function?
+                    let picking_data_without_padding =
+                        picking_rect_info.row_info.remove_padding(data);
```

Comment:
I feel already that we should have a helper layer between the user and `gpu_readback_belt`. If every users need to store the original rectangles so that they can remove the padding, then there should be a helper layer that does all that for us

Reply:
Yeah that's what I meant with the comment :)

Reply:
although, not having the user store the originals is kinda hard I think. I didn't want to do dynamic dispatching on the level of the readbackbelt, i.e. not categorizing all the possible kinds of readbacks on the re_renderer side